### ISSUE TITLE
Issues with client certificates

### ISSUE THREAD

**Issue by adocampo** [CoffeeITWorks/ansible_burp2_client#30]
Hello, I'm having a weird issue, and I don't know how to tackle it... I must admit certificates authorities are quite a mystery for me, I know the fundamentals, but when a problem arises like know, I'm completely lost.

I have installed more than 60 clients without problems, all of them are the exact same version of ubuntu (20.04) and they are working all just perfect... well, all except one.

Playbook finishes apparently without errors but I cannot see on the `/etc/burp/` directory  anything but this:
```
# ls -al
total 16
drwxr-xr-x   3 root root 4096 Mar 10 18:58 .
drwxr-xr-x 108 root root 4096 Mar 10 18:37 ..
drwxr-xr-x   2 root root 4096 Mar 10 18:37 CA-client
-rw-r--r--   1 root root  666 Mar 10 18:37 burp.conf
```
I've reviewed burp.conf and even compared with `diff` with another burp.conf client on any working node and it's ok. On the server, I've make sure I have the config file for that node on `/etc/burp/clientconfdir` and the parameters (password mainly) are just like the other working nodes.

If I run `burp -a b` 
```
# burp -ab
2021-03-10 18:59:23 +0100: burp[595545] Could not find ssl_cert /etc/burp/ssl_cert-client.pem: No such file or directory
2021-03-10 18:59:23 +0100: burp[595545] Could not find ssl_key /etc/burp/ssl_cert-client.key: No such file or directory
2021-03-10 18:59:23 +0100: burp[595545] Could not find ssl_cert_ca /etc/burp/ssl_cert_ca.pem: No such file or directory
2021-03-10 18:59:23 +0100: burp[595545] Connecting to bck01.example.com:4971
2021-03-10 18:59:24 +0100: burp[595545] unexpected command in authorise_client(): e:001D:unable to authorise on server
```
On other places I saw perhaps there is no connection with the server, but I can ping and telnet to the 4971 port just fine. I'm using a non-firewalled and dedicated LAN just for this purpose, so no firewall is involved either.

So, I don't know where to go now. I don't know why didn't create the certificates and why it can't connect to the server...

Could anyone shred some light, please?

**Comment by adocampo** [comment]
Nevermind, I finally managed to make it work.
If anyone falls here, make sure your hostname set on `/etc/burp/burp.conf` matches the host on `clientconfdir`.
I've deployed those nodes with Chef, and it didn't set the hostname properly, even if it was shown just fine on the prompt. So when the ansible playbook generated `burp.conf` it took the wrong name (in my case, the hostname instead the FQDN hostname)
(I hate chef, I'm using it because of an imposition, but I'm slowly migrating everything to ansible)
